server/middleware/measure.go

```diff
@@ -117,37 +117,51 @@ func measureStream() grpc.StreamServerInterceptor {
 }
 
 func (w *wrappedStream) RecvMsg(m interface{}) error {
+	err := w.ServerStream.RecvMsg(m)
+	if err != nil {
+		ulog.E(err)
+	}
 	if w.measurement == nil {
-		err := w.ServerStream.RecvMsg(m)
 		return err
 	}
-	db, coll := request.GetDbAndColl(m)
-	reqMetadata, err := request.GetRequestMetadataFromContext(w.WrappedContext)
-	if err != nil {
-		return err
+
+	if len(w.measurement.GetDBCollTags()) == 0 {
+		// The request is not tagged yet with db and collection, need to do it on the first message
+		db, coll := request.GetDbAndColl(m)
+		reqMetadata, err := request.GetRequestMetadataFromContext(w.WrappedContext)
+		if err != nil {
+			ulog.E(err)
```

Comment:
if err is not nil can we still safely tag it in line `135`?

Reply:
Right, we can't, we need to return the error here and maybe have a log message that this request will not be measured.